Tests/Services/PactContractTests2.swift

```diff
@@ -0,0 +1,155 @@
+    }
+
+    pact = try Pact(consumer: "sanity-consumer", provider: "sanity-provider")
+      .withSpecification(.v4)
+    
+    let config = PactBuilder.Config(pactDirectory: "/tmp/pacts")
+    builder = PactBuilder(pact: pact, config: config)
+    session = URLSession(configuration: .ephemeral)
+  }
+  
+  func test_object() async throws {
+    try builder
+      .uponReceiving("a request for an object")
+      .given("some state")
+      .withRequest(method: .GET, path: "/object")
+      .willRespond(with: 200) { response in
+        try response.jsonBody(
+          .eachLike([
+            "id": .integer(1),
+            "name": .like("Test Object")
+          ])
+        )
+      }
+    
+    try await builder.verify { context in
+      let url = URL(string: "\(context.mockServerURL)/object")!
+      _ = try await self.session.data(from: url)
+    }
+    
+    try pact.writePactFile()
+    
+    let firstInteraction = try XCTUnwrap(getInteractions().first as? [String: Any])
+    
+    let response = try XCTUnwrap(firstInteraction["response"] as? [String: Any])
+
+    assertInlineSnapshot(of: response, as: .json) {
+      #"""
+      {
+        "body" : {
+          "content" : [
+            {
+              "id" : 1,
+              "name" : "Test Object"
+            }
+          ],
+          "contentType" : "application\/json",
+          "encoded" : false
+        },
+        "headers" : {
+          "Content-Type" : [
+            "application\/json"
+          ]
+        },
+        "matchingRules" : {
+          "body" : {
+            "$" : {
+              "combine" : "AND",
+              "matchers" : [
+                {
+                  "match" : "type"
+                }
+              ]
+            },
+            "$[*].id" : {
+              "combine" : "AND",
+              "matchers" : [
+                {
+                  "match" : "integer"
+                }
+              ]
+            },
+            "$[*].name" : {
+              "combine" : "AND",
+              "matchers" : [
+                {
+                  "match" : "type"
+                }
+              ]
+            }
+          }
+        },
+        "status" : 200
+      }
+      """#
+    }
+  }
+  
+  func test_object2() async throws {
+    try builder
+      .uponReceiving("a request for another object")
+      .given("some state")
+      .withRequest(method: .GET, path: "/object2")
+      .willRespond(with: 200) { response in
+        try response.jsonBody(
+          .eachLike([
+            "id": .integer(1),
+            "name": .like("Test Object")
+          ])
+        )
+      }
+    
+    try await builder.verify { context in
+      let url = URL(string: "\(context.mockServerURL)/object2")!
+      _ = try await self.session.data(from: url)
+    }
+    
+    try pact.writePactFile()
+    
+    let interactions = try getInteractions()
+    
+    XCTAssertEqual(1, interactions.count)
+  }
+}
+
+
+extension PactContractTests2 {
```

Comment:
suggestion: Could `private extension` this one and maybe split it with a `// MARK: - Private` comment above it. This way it helps in your minimap:

![Screenshot 2024-05-03 at 9 45 35 AM](https://github.com/jasunde/PactSwift/assets/1733327/be1a0d8b-83ec-4137-8e81-20e24737f2b3)

If you think this could be a useful re-usable utillity in other tests as well, maybe extract it into a "base test" that you then inherit from or a protocol with default implementation you conform your test class to.